services/graph/pkg/config/config.go

```diff
@@ -98,6 +98,7 @@ type Identity struct {
 type API struct {
 	GroupMembersPatchLimit int    `yaml:"group_members_patch_limit" env:"GRAPH_GROUP_MEMBERS_PATCH_LIMIT" desc:"The amount of group members allowed to be added with a single patch request."`
 	UsernameMatch          string `yaml:"graph_username_match" env:"GRAPH_USERNAME_MATCH" desc:"Option to allow legacy usernames. Supported options are 'default' and 'none'."`
+	AssignDefaultUserRole  bool   `yaml:"graph_assign_default_user_role" env:"GRAPH_ASSIGN_DEFAULT_USER_ROLE" desc:"Whether to assign newly created users the default role 'User'. Set this to 'false' if you want to assign roles manually or if the role assignment should happen at first login."`
```

Comment:
```suggestion
	AssignDefaultUserRole  bool   `yaml:"graph_assign_default_user_role" env:"GRAPH_ASSIGN_DEFAULT_USER_ROLE" desc:"Whether to assign newly created users the default role 'User'. Set this to 'false' if you want to assign roles manually, or to 'true' if the role assignment should happen at first login."`
```
Looks good, only a small change.
Q: is this a valid config when using LDAP? (maybe a stupid question)

Reply:
Yeah. This can be used independent of the user backend. There are currently two different ways how the users would get their default role assignment in case AssignDefaultUserRole is set to `false`:

* the admin can manually assign a role using the admin ui
* if during first login the user does not have a role assigned, it will automatically get the "user" role assigned

A third option to allow role assignments based on OIDC claims is currently in the works.